### PR TITLE
Mouse events into xlib

### PR DESCRIPTION
**Pull request by muddasheep**
Use xlib to click on events, use scrollIntoView() to show objects, hook up to signals "event" and "event-after" to be able to wait for click/motion events.

### CODE REVIEW

File: lib/WWW/WebKit2.pm [hunk 1]
```diff
@@ -327,6 +359,12 @@ sub is_loading {
     return 0;
 }
 
+sub handle_form_submission {
+    my ($self, $view, $request) = @_;
```
Comment (jscarty) [inline lib/WWW/WebKit2.pm:363]: Is view needed here?

File: lib/WWW/WebKit2/MouseInput.pm [hunk 2]
```diff
@@ -72,14 +72,58 @@ sub click {
...
+    $self->clear_events;
+    $self->release_mouse_button(1);
+    $self->wait_for_mouse_click_event;
+    $self->pause(100);
+
+    return 1;
+}
+
+sub wait_for_mouse_motion_event {
+    my ($self, $mouse_click) = @_;
```
Comment (jscarty) [inline lib/WWW/WebKit2/MouseInput.pm]: Is $mouse_click needed?

File: lib/WWW/WebKit2/MouseInput.pm [hunk 3]
```diff
@@ -72,14 +72,58 @@ sub click {
...
+    $self->wait_for_condition(sub {
+        $self->find_event(sub {
+            my $event_type = scalar $_;
+            return $event_type =~ /EventMotion/;
+        });
+    });
+}
+
+sub wait_for_mouse_click_event {
+    my ($self, $mouse_click) = @_;
```
Comment (jscarty) [inline lib/WWW/WebKit2/MouseInput.pm]: is $mouse_click needed here as well?

**Review by jscarty — CHANGES_REQUESTED** [review]
Looks good overall. Just need to remove some unneeded variables and lines.

File: lib/WWW/WebKit2.pm (hunk 1 above)
Comment (muddasheep) [reply lib/WWW/WebKit2.pm:363]: I will leave $view here to show which variable is given to handle_form_submission from webkit.

Comment (jscarty) [reply lib/WWW/WebKit2.pm:363]: Fine with me

**Review by jscarty — APPROVED** [review]
Ship it!

**Pull request merged**